.github/workflows/check-links-book.yaml

```diff
@@ -0,0 +1,67 @@
+# We also have a check-links-markdown job, however it will not spot mdbook
+# mistakes such as forgetting to list an .md file in SUMMARY.md.
+# Running a link checker on the generated HTML is more reliable.
+
+name: check-links-book
+
+on:
+  workflow_call:
+
+env:
+  RUSTFLAGS: "-C debuginfo=0"
+
+concurrency:
+  # See notes in `pull-request.yaml`
+  group: ${{ github.workflow }}-${{ github.ref }}-check-links
+  cancel-in-progress: true
+
+jobs:
+  changes:
+    runs-on: ubuntu-latest
+    permissions:
+      pull-requests: read
+    outputs:
+      src: ${{ steps.filter.outputs.src }}
+    steps:
+      - uses: dorny/paths-filter@v2
+        id: filter
+        with:
+          filters: |
+            src:
+              - '.github/workflows/check-links-book.yaml'
+              - 'web/book/**'
+
+  check:
+    needs: changes
+    if: ${{ needs.changes.outputs.src == 'true' }}
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - uses: baptiste0928/cargo-install@v2
+        with:
+          crate: mdbook
+
+      # the link checker
+      - uses: baptiste0928/cargo-install@v2
+        with:
+          crate: hyperlink
+
+      - name: Cache
+        uses: Swatinem/rust-cache@v2
+        with:
+          prefix-key: 0.8.1
```

Comment:
FYI, prefix-key should be updated by the Cargo.toml.

https://github.com/PRQL/prql/blob/0224b6efb140b8aec4d8c85bd62967cebdf7b2e2/crates/prql_compiler/Cargo.toml#L78-L85

Reply:
Ah, missed that, thanks @eitsupi